src/state/opamSysInteract.ml

```diff
@@ -621,3 +621,18 @@ let update () =
   | Some (cmd, args) ->
     try sudo_run_command cmd args
     with Failure msg -> failwith ("System package update " ^ msg)
+
+let repo_enablers () =
+  if family () = Centos then
+    let (needed, _) =
+      packages_status (OpamSysPkg.raw_set (OpamStd.String.Set.singleton "epel-release"))
+    in
+    if OpamSysPkg.Set.is_empty needed then
+      None
+    else
+      Some "On CentOS/RHEL, many packages may assume that the Extra Packages \
+            for Enterprise Linux (EPEL) repository has been enabled. This is \
+            typically done by installing the 'epel-release' package. Please \
+            see https://fedoraproject.org/wiki/EPEL for more information"
```

Comment:
We're going to need a number of these workarounds for different operating systems (e.g. Alpine's edgecommunity, Brew's custom taps and so on).  Would it be feasible to move this into a package `{message}` entry somehow rather than hardcoding Fedora-specific logic in the client?  Obviously, other opam-repositories other than OCaml's may not make this assumption as well.

Reply:
The logic here is being embedded in the section of the opam client which deals with the logic for specific package managers, so I think it is in the right place for now?

It's not feasible between now and releasing 2.1, no. At the moment, for EPEL specifically, this is fixing a regression in behaviour because of a poor decision in 2017 to handle this in opam-depext! All this PR is doing (the actual new code is only a tiny bit on an error message) is displaying this message if it thinks it _might_ be helpful. We should definitely follow this up with better (and generic) tagging, or some such, of depexts, though.

My understanding is that the Alpine repos work much better because the error message from apk clearly explains what needs doing (i.e. this PR brings dnf up to apk's standard 🙂). ppas, custom taps, etc. would be a _completely_ new feature - EPEL is an official repository system (as, again, if I understand correctly, edgecommunity is an official repo, it's just not necessarily enabled by default?).